mantidimaging/gui/windows/spectrum_viewer/presenter.py

```diff
@@ -297,7 +294,8 @@ def handle_roi_moved(self) -> None:
             "spec_mode": self.spectrum_mode,
             "shutter_norm": self.view.shuttercount_norm_enabled(),
             "chunk_start": chunk_start,
-            "chunk_end": chunk_end
+            "chunk_end": chunk_end,
+            "spectrum_data_dict": self.view.spectrum_widget.spectrum_data_dict
```

Comment:
This goes a bit against my intuition. `thread_finished_processing` should be running back in the main thread (because its triggered via a signal). And it needs to be updating the real cache. So using `self.view` there should be fine.